updater/library/src/updater.rs

```diff
@@ -125,6 +134,39 @@ fn update_internal(config: &ResolvedConfig) -> anyhow::Result<UpdateStatus> {
     return Ok(UpdateStatus::UpdateInstalled);
 }
 
+fn inflate(patch_path: &Path, base_path: &Path, output_path: &Path) -> anyhow::Result<()> {
+    info!("Patch is compressed, inflating...");
+    use comde::de::Decompressor;
+    use comde::zstd::ZstdDecompressor;
+    use std::fs::File;
+    use std::io::{BufReader, BufWriter};
+
+    // Open all our files first for error clarity.  Otherwise we might see
+    // PipeReader/Writer errors instead of file open errors.
+    info!("Reading base file: {:?}", base_path);
+    let base_r = File::open(base_path)?;
+    let compressed_patch_r = BufReader::new(File::open(patch_path)?);
+    let output_file_w = File::create(&output_path)?;
+
+    let (patch_r, patch_w) = pipe::pipe();
+
+    let decompress = ZstdDecompressor::new();
+    std::thread::spawn(move || {
+        let result = decompress.copy(compressed_patch_r, patch_w);
+        // If this thread fails, undoubtedly the main thread will fail too.
+        // Most important is to not crash.
+        if let Err(err) = result {
+            error!("Decompression thread failed: {err}");
```

Comment:
Rust noob question:
What does this `error!(...)` do? Does it propagate as the returned `Result` or are we swallowing it and always returning `Ok`?